src/main/java/core/basesyntax/service/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.stategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy1) {
```

Comment:
The parameter name 'operationStrategy1' should be renamed to 'operationStrategy' to match the field name and improve readability.